### PR TITLE
(DOCUMENT-448) Specify format for 'ensure' value when using a version.

### PR DESCRIPTION
**Pull request by rnelson0**
https://tickets.puppetlabs.com/browse/DOCUMENT-448

### CODE REVIEW

File: lib/puppet/type/package.rb [hunk 1]
```diff
@@ -74,6 +74,11 @@ module Puppet
         value. On packaging systems that manage configuration files separately
         from "normal" system files, you can uninstall config files by
         specifying `purged` as the ensure value. This defaults to `installed`.
+
+        Version numbers must match the full version to install, including
+        release if the provider uses a release moniker. Ranges or semver
+        patterns are not accepted. E.g. to install the rpm
```
Comment (gguillotte) [inline lib/puppet/type/package.rb]: For clarity, I'd replace "E.g." with "For example,".

**Comment by gguillotte** [comment]
Travis is failing because the commit starts with a DOCUMENT project ticket. Can you amend the commit to replace (DOCUMENT-448) with (docs), and note the ticket number in the commit description instead?

One other suggestion, then :+1: from me.

**Comment by rnelson0** [comment]
@gguillotte Done and done! Just waiting on travis.

**Comment by rnelson0** [comment]
Yay, tests pass this time!

File: lib/puppet/type/package.rb [hunk 2]
```diff
@@ -74,6 +74,11 @@ module Puppet
         value. On packaging systems that manage configuration files separately
         from "normal" system files, you can uninstall config files by
         specifying `purged` as the ensure value. This defaults to `installed`.
+
+        Version numbers must match the full version to install, including
+        release if the provider uses a release moniker. Ranges or semver
+        patterns are not accepted. For example, to install the rpm
```
Comment (joshcooper) [inline lib/puppet/type/package.rb]: minor nit, the `gem` package provider does support ranges and semver constraints. This was added in https://tickets.puppetlabs.com/browse/PUP-4131. AFAIK none of the other providers support them.

Comment (gguillotte) [inline lib/puppet/type/package.rb]: Should we note provider-specific exceptions in the type reference? Or couch that statement with something along the lines of "... unless the package provider explicitly allows it"?

Comment (joshcooper) [inline lib/puppet/type/package.rb]: not sure... for comparison the `source` parameter does https://github.com/puppetlabs/puppet/blob/master/lib/puppet/type/package.rb#L360-L363

**Comment by rnelson0** [comment]
I added a second commit to address the `gem` provider. If this is better, let me know and I'll rebase, otherwise I'll ditch it in favor of something better.

**Comment by gguillotte** [comment]
@rnelson0 @joshcooper If `gem` is the only exception, then I think all that needs to change over the original commit is to mention that as briefly as possible: "Ranges or semver patterns are not accepted **except for the `gem` package provider**."

Otherwise I'm :+1: for rebasing and merging this..

**Comment by rnelson0** [comment]
Okay, I think I got this now!

**Comment by gguillotte** [comment]
I'm good with this. I can't seem to merge it, though, GH is saying the branch is out of date with `master`. :/ I'm having problems with a different PR that make me think there might be an issue on my end or with GH. CC @nfagerlund

**Comment by gguillotte** [comment]
Got the merge button and merged it! Thanks @rnelson0!

**Pull request merged**